src/mpid/common/shm/mpidu_shm_alloc.c

```diff
@@ -213,59 +166,21 @@ int MPIDU_shm_seg_commit(MPIDU_shm_seg_t * memory, MPIDU_shm_barrier_t ** barrie
             serialized_hnd_size = strlen(serialized_hnd);
             MPIR_Assert(serialized_hnd_size < MPIR_pmi_max_val_size());
 
-            /* must come before barrier_init since we use OPA in that function */
-#ifdef OPA_USE_LOCK_BASED_PRIMITIVES
-            ipc_lock = (OPA_emulation_ipl_t *) ((char *) memory->base_addr + ipc_lock_offset);
-            ret = OPA_Interprocess_lock_init(ipc_lock, local_rank == 0);
-            MPIR_ERR_CHKANDJUMP1(ret != 0, mpi_errno, MPI_ERR_OTHER, "**fail", "**fail %d", ret);
-#endif
-
-            mpi_errno =
-                MPIDU_shm_barrier_init((MPIDU_shm_barrier_t *) memory->base_addr, barrier, TRUE);
-            MPIR_ERR_CHECK(mpi_errno);
+            MPIDU_Init_shm_put(serialized_hnd, MPIDU_SHM_MAX_FNAME_LEN);
```

Comment:
Or, did you mean to use `serialized_hnd_size` here, and `assert(serialized_hnd_size < MPIDU_SHM_MAX_FNAME_LEN)`?

Reply:
Yes, this is also a leftover of when I rebased on the PMI refactor PR